crypto/packet.c

```diff
@@ -232,6 +265,19 @@ static int wpacket_intern_close(WPACKET *pkt, WPACKET_SUB *sub, int doclose)
                 && !put_value(&buf[sub->packet_len], packlen,
                               sub->lenbytes))
             return 0;
+    } else if (pkt->endfirst && sub->parent != NULL) {
+        size_t tmplen = packlen;
+        size_t numlenbytes = 1;
+
+        while ((tmplen = tmplen >> 8) > 0)
+            numlenbytes++;
+        if (!WPACKET_put_bytes__(pkt, packlen, numlenbytes))
+            return 0;
+        if (packlen > 0x7f) {
+            numlenbytes |= 0x80;
+            if (!WPACKET_put_bytes_u8(pkt, numlenbytes))
+                return 0;
+        }
     }
 
     if (doclose) {
```

Comment:
Ok, `packlen` is being written to the buffer, fine, but then what?  At least in DER terms, it needs to be accumulated in the parent packet, otherwise we end up with too short lengths all over the place.  Remember that the length of a SEQUENCE, SET or CHOICE is the sum of the lengths of all its elements, including their tags and lengths...

Reply:
Never mind this, I was looking at the wrong byte...